builtin/mainmenu/pkgmgr.lua

```diff
@@ -25,27 +25,46 @@ function get_mods(path,retval,modpack)
 			local toadd = {}
 			retval[#retval + 1] = toadd
 
-			local mod_conf = Settings(prefix .. DIR_DELIM .. "mod.conf"):to_table()
-			if mod_conf.name then
-				name = mod_conf.name
+			-- Get config file
+			local mod_conf
+			local modpack_conf = io.open(prefix .. DIR_DELIM .. "modpack.conf")
+			if modpack_conf then
+				toadd.is_modpack = true
+				modpack_conf:close()
+
+				mod_conf = Settings(prefix .. DIR_DELIM .. "modpack.conf"):to_table()
+				if mod_conf.name then
+					name = mod_conf.name
+				end
+			else
+				mod_conf = Settings(prefix .. DIR_DELIM .. "mod.conf"):to_table()
+				if mod_conf.name then
+					name = mod_conf.name
+				end
 			end
 
+			-- Read from config
 			toadd.name = name
 			toadd.author = mod_conf.author
 			toadd.release = tonumber(mod_conf.release or "0")
 			toadd.path = prefix
 			toadd.type = "mod"
 
+			-- Check modpack.txt
+			--  Note: modpack.conf is already checked above
+			local modpackfile = io.open(prefix .. "modpack.txt")
+			if modpackfile then
```

Comment:
~= nil

Reply:
`modpackfile` is either a valid `io` file reference or `nil`. This simple check makes sense here.

Reply:
yeah but it's not consistent with later calls :)